src/pyramid/compat.py

```diff
@@ -0,0 +1,188 @@
+# This is not actually an API and *should not* be used by Pyramid 2.X apps or
+# Pyramid 2.X itself.  But many apps in the wild depend on this stuff, and it's
+# just more pragmatic to just leave it here, even though we no longer support
+# Python 2.
+
+import inspect
+import platform
+import sys
+import types
+
+WIN = platform.system() == 'Windows'
+
+try:  # pragma: no cover
+    import __pypy__
+
+    PYPY = True
+except BaseException:  # pragma: no cover
+    __pypy__ = None
+    PYPY = False
+
+import pickle
+
+from functools import lru_cache
+
+PY3 = sys.version_info[0] == 3
+
+string_types = (str,)
+integer_types = (int,)
+class_types = (type,)
+text_type = str
+binary_type = bytes
+long = int
+
+
+def text_(s, encoding='latin-1', errors='strict'):  # pragma: no cover
+    """ If ``s`` is an instance of ``binary_type``, return
+    ``s.decode(encoding, errors)``, otherwise return ``s``"""
+    if isinstance(s, binary_type):
+        return s.decode(encoding, errors)
+    return s
```

Comment:
These functions are already defined in `pyramid.util`. Please import them from there and re-export them instead of re-defining them here.